xarray/tests/test_dask.py

```diff
@@ -1035,6 +1035,13 @@ def test_unify_chunks_shallow_copy(obj, transform):
     assert_identical(obj, unified) and obj is not obj.unify_chunks()
 
 
+@pytest.mark.parametrize("obj", [make_da()])
+def test_auto_chunk_da(obj):
+    expected = obj.chunk("auto").data
+    actual = obj.data.rechunk("auto")
+    np.testing.assert_array_equal(actual, expected)
```

Comment:
we should check that the chunk sizes are equal so `assert actual.chunks == expected.chunks`?  Not sure a dataset test is really necessary since both dataarrays and datasets go through the same code path.

Reply:
Added!